molecule_libvirt/playbooks/create.yml

```diff
@@ -0,0 +1,259 @@
+        uri: "{{ 'qemu+ssh://'+item.libvirt_user+'@'+item.libvirt_host+'/system' if item.libvirt_host is defined else libvirt_uri }}"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      when: item.network_name is not defined
+
+    - name: Get qcow2 image
+      get_url:
+        url: "{{ image_url | default(default_image_url) }}"
+        dest: "{{ libvirt_default_image_path }}/{{ item.name }}.qcow2"
+        mode: '0660'
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Generate OpenSSH key pair
+      openssh_keypair:
+        path: "{{ molecule_private_key_path }}"
+        size: 2048
+      register: ssh_key
+      delegate_to: "localhost"
+
+    - name: Create temporary folder
+      file:
+        path: "{{ temp_folder }}/{{ item.name }}-molecule"
+        state: directory
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Create user-data from template
+      template:
+        src: user-data.j2
+        dest: "{{ temp_folder }}/{{ item.name }}-molecule/user-data"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Create meta-data from template
+      template:
+        src: meta-data.j2
+        dest: "{{ temp_folder }}/{{ item.name }}-molecule/meta-data"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Create cloud-init iso file
+      command:
+        "mkisofs -o
+        {{ libvirt_default_image_path }}/{{ item.name }}-cloud-init.iso
+        -V cidata -J -r user-data meta-data"
+      args:
+        chdir: "{{ temp_folder }}/{{ item.name }}-molecule"
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Remove tempfolfer
+      file:
+        path: "{{ temp_folder }}/{{ item.name }}-molecule"
+        state: absent
+      with_items: "{{ molecule_yml.platforms }}"
+      loop_control:
+        label: "{{ item.name }}"
+      delegate_to: "{{ item.libvirt_host | default('localhost') }}"
+
+    - name: Create an image shell with the desired size
+      command:
+        "qemu-img create -f qcow2 -o preallocation=metadata
+        {{ item.name }}-temp.qcow2
+        {{ disk_size | default(default_disk_size ) }}"
+      args:
+        chdir: "{{ libvirt_default_image_path }}"
+      environment:
+        LIBGUESTFS_BACKEND: direct
+      with_items: "{{ molecule_yml.platforms }}"
```

Comment:
Do we want to keep `with_items` or do we want to use `loop`?

Reply:
with_items has been changed for loop.